client/src/main/java/com/microsoft/durabletask/DurableTaskGrpcClient.java

```diff
@@ -155,12 +156,21 @@ public OrchestrationMetadata waitForInstanceStart(String instanceId, Duration ti
         TaskHubSidecarServiceBlockingStub grpcClient = this.sidecarClient.withDeadlineAfter(
                 timeout.toMillis(),
                 TimeUnit.MILLISECONDS);
-        GetInstanceResponse response = grpcClient.waitForInstanceStart(request);
+
+        GetInstanceResponse response;
+        try {
+            response = grpcClient.waitForInstanceStart(request);
+        } catch (StatusRuntimeException e) {
+            if (e.getStatus().getCode() == Status.Code.DEADLINE_EXCEEDED) {
+                throw new TimeoutException("Start orchestration timeout reached");
```

Comment:
I'm not sure if there's a Java convention for this, but I usually try to end simple exception messages like this with a period.
```suggestion
                throw new TimeoutException("Start orchestration timeout reached.");
```